inst/tests/tests.Rraw

```diff
@@ -12428,6 +12428,10 @@ DTout = data.table(
 )
 test(1866.6, melt(DT, measure.vars = patterns("^x", "^y", cols=names(DT))), DTout)
 
+# informative errors for bad user-provided cols arg to patterns
+test(1866.80, melt(data.table(x1=1,x2=2,y1=3,y2=4),measure.vars=patterns("2",cols=NULL)), error="cols must be a character vector of column names")
```

Comment:
I think the test is easier to read if we factor out the repeated part so the reader can focus on what changes from .80 to .81